app/assets/javascripts/components/Map.js

```diff
@@ -75,10 +75,12 @@ class Map extends React.Component {
   }
 
   _popUpFor(layer) {
+    // debugger
     return `<h1>${layer.name}</h1>
             <p>${layer.description}</p>
             <div class="link-group">
-              <a href="/tasks/${layer.id}">Learn more</a>
+              <a href="/campaigns/${layer.events ? layer.events[0].campaign_id : this.props.campaignId}">Campaign</a>
```

Comment:
I have discussed with Juan Carlos and will try to find a design solution. With regards to responsive maps fitting to screen height, we have the use case for events that there is a header above the map therefore on mobile map fitting is not possible. I can approximate but not force map to fit to bottom of screen in all cases.